### ISSUE TITLE
spec: MVP for Blocksync

### ISSUE THREAD

**Issue by josef-widder** [circlefin/malachite#425]
In order to stabilize the system in case nodes fall behind, we want to have a synchronization protocol that allows reasonably quick catch-up.

Main principles

- It **always** runs in parallel with consensus
    - this avoids the problems of switching between blocksync and consensus that are present in CometBFT
- It uses a specific 1-to-1 communication with peers (p2p)
    - For publishing status we should use localcast, namely, we publish it to all connected peers
- The logic for deciding is within consensus, and doesn't change
    - protocol retrieves blocks, and commits that can be fast-tracked to consensus re-using the existing logic (cf. #408)

For the MVP we consider the following obvious simplifications:
- there is no equivocation nor (dangerous) Byzantine behavior
- the protocol does not validate signatures (just forward to node/driver)
- the protocol selects peers to download from at random (no load balancing)
- no DDoS protection (BAR model approach)
- periodic announcement of base height and current height (+ potential whether a certificate or a value is already known for that height)

Open questions:
- What parts should run concurrently to the driver
- the frequency of announcements
- downloading of blocks in parallel (multiple height) or sequentially
    - recall that we don't do signature verification in the first version
    - for that we would need to know the validator set of the requested heights within the block sync protocol

Definition of done:
- [x] #546
- [x] #559
- [x] #588
- [ ] Stretch goal: performance considerations (timeouts, parallelism, load balancing, DDos, etc)
